astropy/wcs/utils.py

```diff
@@ -118,6 +142,35 @@ def _wcs_to_celestial_frame_builtin(wcs):
                 representation_type=SphericalRepresentation,
                 obstime=wcs.wcs.dateobs or None,
             )
+        elif (
+            "LN" in xcoord
+            and "LT" in ycoord
+            and "H" not in xcoord
+            and "CR" not in xcoord
+        ):
+            if Version(_wcs.__version__) >= Version("8.0"):
+                a_radius = wcs.wcs.aux.a_radius
+                b_radius = wcs.wcs.aux.b_radius
+                c_radius = wcs.wcs.aux.c_radius
```

Comment:
If we need the `if`, then there should be an `else` that raises an error that the wcslib version does not support this.